### PR TITLE
Fix babel cache bleed

### LINKED ISSUE #568: Specifying `commonjs` as one of the output formats forces generators to compile to regenerator runtime
`--format modern,es,umd`

generators are left untouched

`--format modern,es,umd,cjs`

transpiles generators for all formats, no matter where `cjs` is inserted.

Would be happy to make a fix if you can point me in the right direction.

### PR DESCRIPTION
**Pull request by developit**
Instantiating babel.custom() separately for each output should fix #568, which I believe is caused by cached configuration bleeding from `cjs` into other output targets.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -467,7 +467,7 @@ function createConfig(options, entry, format, writeMeta) {
 
 	const externalPredicate = new RegExp(`^(${external.join('|')})($|/)`);
 	const externalTest =
-		external.length === 0 ? () => false : id => externalPredicate.test(id);
+		external.length === 0 ? id => false : id => externalPredicate.test(id);
```
Comment (ForsakenHarmony) [inline src/index.js]: does this affect anything?

File: src/index.js [hunk 2]
```diff
@@ -630,9 +631,11 @@ function createConfig(options, entry, format, writeMeta) {
...
 							config._sizeInfo = Promise.all(
-								Object.values(bundle).map(({ code, fileName }) =>
-									code ? getSizeInfo(code, fileName, options.raw) : false,
-								),
+								Object.values(bundle).map(({ code, fileName }) => {
+									if (code) {
+										return getSizeInfo(code, fileName, options.raw);
+									}
+								}),
 							).then(results => results.filter(Boolean).join('\n'));
```
Comment (ForsakenHarmony) [inline src/index.js]: I guess the implicit undefined return will work :thinking:

File: src/index.js [hunk 3]
```diff
@@ -622,6 +622,7 @@ function createConfig(options, entry, format, writeMeta) {
 									fs.writeFile(
 										getNameCachePath(),
 										JSON.stringify(nameCache, null, 2),
+										() => {},
```
Comment (ForsakenHarmony) [inline src/index.js]: same as above, does this change the behavior?

File: src/index.js (hunk 1 above)
Comment (developit) [reply src/index.js]: no, just typescript was yelling at me about it.

File: src/index.js (hunk 3 above)
Comment (developit) [reply src/index.js]: same deal, was a typescript warning

File: src/index.js (hunk 2 above)
Comment (developit) [reply src/index.js]: yeah cause we cast to boolean on L639. this was another TS warning lol

**Comment by ForsakenHarmony** [comment]
bors retry

**Comment by ForsakenHarmony** [comment]
bors retry

**Comment by ForsakenHarmony** [comment]
bors retry

**Pull request merged**